docs/usage/config-overview.md

```diff
@@ -0,0 +1,250 @@
+These default values (or absence of) are documented in the Renovate Docs website.
+For example:
+
+- The default value for `onboarding` is `true`
+- The option `labels` has no default value, meaning no labels are added to PRs by default
+
+Default config is loaded first, and then is superseded/overridden by the configuration types below.
+
+### Global config
+
+Global config is the term for config defined by the person or team responsible for running the bot.
+Sometimes it's referred to as "bot config" because it's the config passed to the bot by the person running it.
+Global config can contain config which is "global only" as well as any configuration options which are valid in Inherited config or Repository config.
+
+If you are an end user of Renovate, such as a user of the Mend Renovate App, then you don't need to care too much about global config other than knowing that there are some settings you cannot change because they are global-only.
+You can skip the rest of this "Global config" section and proceed to "Inherited config".
+
+Global config can be read from file, environment, or CLI parameters.
+You must configure at least one of these in order to provide your bot with the minimum details it needs to run, e.g. credentials.
+
+#### File config
+
+The first place Renovate global config is read from is from file.
+By default Renovate checks for the existence of `config.js` in the current working directory, but you can override this by defining `RENOVATE_CONFIG_FILE` in env, e.g. `RENOVATE_CONFIG_FILE=/tmp/my-renovate-config.js`.
+
+By default Renovate allows the config file to be _missing_ and does not error if it cannot find it, however if you have configured `RENOVATE_CONFIG_FILE` but the path you specified is not found then Renovate will error and exit, as it's assumed you have a configuration problem.
+If the file is found but cannot be parsed then Renovate will also error and exit.
+
+Global config files can be `.js` or `.json` files.
+JS files can use synchronous or asyncronous methods inside, including even to fetch config information from remote hosts.
+
+#### Environment config
+
+Global config can be defined using environment variables.
+Each config option which is supported in env has the prefix `RENOVATE_`.
+For example, `RENOVATE_PLATFORM=gitlab` is the same as setting `"platform": "gitlab"` in File config.
+
+Although the mapping from configuration option name to its corresponding Environment config name is fairly easy to understand, we recommend you consult the documentation for the field `env` for each option.
+If the `env` field is missing then it means that the configuration option does not have its own Environment config variable name.
+
+A special case for Environment config is the `RENOVATE_CONFIG` "meta" config option.
+The `RENOVATE_CONFIG` option accepts a stringified full config, e.g. `RENOVATE_CONFIG={"platform":"gitlab","onboarding":false}`.
```

Comment:
I think we should add some samples for the different types, especially arrays are often a confusing issue when it comes to env vars.

Reply:
Added, see below